From c3f484890012370c900676f4d553022001a8fcef Mon Sep 17 00:00:00 2001
From: Gaosheng Cui <cuigaosheng1@huawei.com>
Date: Thu, 22 Sep 2022 16:38:55 +0800
Subject: [PATCH 04/36] [Backport] neighbour: Remove unused inline function
 neigh_key_eq16()

stable inclusion
from stable-v5.10.184
commit 53076071fb92a46119ba7f4e3369f1e7a5237f9c
category: bugfix
bugzilla: 188948
CVE: CVE-2023-3338

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=53076071fb92a46119ba7f4e3369f1e7a5237f9c

---------------------------

commit c8f01a4a54473f88f8cc0d9046ec9eb5a99815d5 upstream.

All uses of neigh_key_eq16() have
been removed since commit 1202cdd66531 ("Remove DECnet support
from kernel"), so remove it.

Signed-off-by: Gaosheng Cui <cuigaosheng1@huawei.com>
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Dong Chenchen <dongchenchen2@huawei.com>
---
 include/net/neighbour.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/net/neighbour.h b/include/net/neighbour.h
index 478d49d45914..8e4404efd910 100644
--- a/include/net/neighbour.h
+++ b/include/net/neighbour.h
@@ -268,11 +268,6 @@ static inline void *neighbour_priv(const struct neighbour *n)
 
 extern const struct nla_policy nda_policy[];
 
-static inline bool neigh_key_eq16(const struct neighbour *n, const void *pkey)
-{
-	return *(const u16 *)n->primary_key == *(const u16 *)pkey;
-}
-
 static inline bool neigh_key_eq32(const struct neighbour *n, const void *pkey)
 {
 	return *(const u32 *)n->primary_key == *(const u32 *)pkey;
-- 
2.17.1

